### DISCUSSION TITLE
Python 3.7 image not available

### DISCUSSION THREAD

**Discussion by xtineskim** [renovatebot/renovate#14588] [Request Help]
Looking in my debug logs, saw this. Docker image that's pulled instead is latest 3.10
```
WARN: Failed to find a tag satisfying constraint, using "latest" tag instead(branch="renovate/itsdangerous-2.x")
{
  "depName": "docker.io/renovate/python",
  "constraint": "3.7",
  "scheme": "pep440"
}
```

**Comment by rarkins** [comment]
Where does that constraint come from, do you know?

**Reply by xtineskim** [reply]
Its specified in my renovate.json file:
```
{
  "extends": [
    "config:base"
  ],
  "pip-compile": {
    "enabled": true,
    "fileMatch": ["(^|/)requirements\\.in$"]
  },
  "constraints": {
    "python": "3.7"
  }
}
```

**Reply by rarkins** [reply]
Can you try a pep440 constraint, Eg `~=3.7.0`?

**Reply by xtineskim** [reply]
I just set it, and retried one of my open PRs.
Still getting the same output of :
```
WARN: Failed to find a tag satisfying constraint, using "latest" tag instead(branch="renovate/click-8.x")
{
  "depName": "docker.io/renovate/python",
  "constraint": "3.7.0",
  "scheme": "pep440"
}
```

**Reply by rarkins** [reply]
It exists: https://hub.docker.com/layers/renovate/python/3.7.0/images/sha256-ecd3cb21e09391ff44a8d4fdb996fbbe336d8c0fbef2d61853fc0752b113a9e6?context=explore

Can you create a reproduction repo to demonstrate?

**Reply by xtineskim** [reply]
yup! https://github.com/ckim328/repro-boa/pull/4
thanks for the quick responses too 👍

**Reply by rarkins** [reply]
I think if you try `~=3.7.0` then it will work
